airbyte/mcp/server.py

```diff
@@ -0,0 +1,341 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+
+"""PyAirbyte MCP Server implementation."""
+
+from __future__ import annotations
+
+import json
+import os
+from typing import Any
+
+import mcp.server.stdio
+from mcp import types
+from mcp.server.lowlevel import Server
+
+from airbyte import get_source
+from airbyte.caches.util import get_default_cache
+from airbyte.sources import get_available_connectors
+
+
+app = Server("pyairbyte-mcp")
+
+
+def _detect_hardcoded_secrets(config: dict[str, Any], spec: dict[str, Any]) -> list[str]:
+    """Detect hardcoded secrets in config that should be environment variables."""
+    hardcoded_secrets = []
+
+    properties = spec.get("properties", {})
+
+    for field_name, field_value in config.items():
+        if field_name not in properties:
+            continue
+
+        field_spec = properties[field_name]
+
+        is_secret = (
+            field_spec.get("format") == "password"
+            or field_spec.get("writeOnly") is True
+            or "password" in field_name.lower()
+            or "secret" in field_name.lower()
+            or "token" in field_name.lower()
+            or "key" in field_name.lower()
+        )
+
+        if (
+            is_secret
+            and isinstance(field_value, str)
+            and not (field_value.startswith("${") and field_value.endswith("}"))
+            and field_value not in os.environ
+        ):
+            hardcoded_secrets.append(field_name)
+
+    return hardcoded_secrets
+
+
+def _generate_config_markdown(connector_name: str, spec: dict[str, Any]) -> str:
+    """Generate markdown documentation for connector configuration."""
+    properties = spec.get("properties", {})
+    required = spec.get("required", [])
```

Comment:
There's already a built method that does this. Don't re-build from scratch - wrap the existing instead. I think it is source.get_config_spec_markdown, or similar. There's also a "print" version of it, which would send with rich to terminal - but I don't think we want to call that version. (We just want to call what it calls.)